src/backend/schemas/context.py

```diff
@@ -0,0 +1,154 @@
+from typing import Optional
+
+from fastapi import Request
+from pydantic import BaseModel, Field
+
+from backend.crud import user as user_crud
+from backend.database_models.database import DBSessionDep
+from backend.schemas.agent import Agent, AgentToolMetadata
+from backend.schemas.metrics import MetricsAgent, MetricsMessageType, MetricsUser
+from backend.schemas.user import User
+from backend.services.utils import get_deployment_config
+
+
+class Context(BaseModel):
+    request: Optional[dict] = {}
+    response: Optional[dict] = {}
+    receive: Optional[dict] = {}
+    trace_id: str = "default"
+    user_id: str = "default"
+    event_type: MetricsMessageType = None
```

Comment:
can this default to `MetricsMessageType.UNKNOWN_SIGNAL`? it'll limit the type to just MetricsMessageType

Reply:
can we also move it down with `metrics_user`
`metrics_agent`

Reply:
actually, I take this back, the check is currently `event_type is None`